src/main/java/org/opentripplanner/common/TurnRestriction.java

```diff
@@ -3,13 +3,14 @@
 import java.io.Serializable;
 
 import org.opentripplanner.routing.core.TraverseModeSet;
+import org.opentripplanner.routing.edgetype.StreetEdge;
 import org.opentripplanner.routing.graph.Edge;
 
 public class TurnRestriction implements Serializable {
     private static final long serialVersionUID = 6072427988268244536L;
     public TurnRestrictionType type;
-    public Edge from;
-    public Edge to;
+    public StreetEdge from;
+    public StreetEdge to;
     public RepeatingTimePeriod time;
     public TraverseModeSet modes;
 
```

Comment:
Could you make as many of these final as possible? That would increase thread safty...

Reply:
I've done that in https://github.com/opentripplanner/OpenTripPlanner/pull/3899/commits/0d4b8f946970ffefea15781f3ba40dde2c981736